File: 프로그래머스/lv1/김서영2/문자열나누기.cpp
#include <string>
#include <vector>

using namespace std;

int solution(string s)
{
    int answer = 0, countDifferent = 0, countSame = 1;
    char x = s[0];
    for (int i = 1; i < s.size(); i++)
    {
        if (s[i] == x)
            countSame++;
        else
            countDifferent++;
        if (countDifferent == countSame)
        {
            answer++;
            x = s[i + 1];
            countDifferent = 0;
            countSame = 0;
        }
    }
    // 남은 문자열 처리
    if (countSame != countDifferent)
        answer++;
    return answer;
}

File: 프로그래머스/lv1/이재표/문자열 나누기.java
import java.util.*;
class Solution {
    public int solution(String s) {
        int answer = 0;
        char x='-';
        int xcnt=0;
        int cnt=0;
        for(int i=0;i<s.length();i++){
            if(xcnt==0){
                x=s.charAt(i);
                xcnt++;
                continue;
            }
            if(s.charAt(i)==x){
                xcnt++;
            }else{
                cnt++;
            }
            if(xcnt==cnt){
                answer++;
                xcnt=0;
                cnt=0;
            }
        }
        if(xcnt!=0||cnt!=0){
            answer++;
        }
        return answer;
    }
}
